blog/views.py

```diff
@@ -0,0 +1,47 @@
+from django.contrib.auth.decorators import login_required
+from django.http import HttpRequest, HttpResponseRedirect
+from django.urls import reverse_lazy
+from django.views import generic
+from blog.models import Post, Commentary
+from blog.forms import CommentaryForm
```

Comment:
Imports should be grouped and separated by a blank line according to standard library, external libraries, and custom imports. This practice enhances both code readability and maintainability. 

Read more about topic [here](https://peps.python.org/pep-0008/#imports)